Added support for WAFV2

eharris91 opened the pull request:
In response to [Support for WAFv2 #16](https://github.com/MikeSouza/serverless-associate-waf/issues/16)
Added two additional config parameters:

- version:  [V2, Regional] 
version defaults to Regional

- scope:  [REGIONAL, CLOUDFRONT]
scope defaults to REGIONAL

MikeSouza commented:
@eharris91 Thank you for your PR.

I've commented about three things, please address these and I think you're good to go.

eharris91 commented:
@MikeSouza Thanks for the feedback, this is my first PR so still trying to figure all this out.

eharris91 commented:
@MikeSouza Based on your feedback:

- The scope parameter logic is removed

- Added tests to index.spec.js, code coverage now at 100%

- Updated documentation in README

MikeSouza commented on README.md:
```diff
@@ -26,11 +26,13 @@ Add your custom configuration:
+| Property | Required | Type     | Default   | Description                                                    |
+|----------|----------|----------|-----------|----------------------------------------------------------------|
+| `name`   |  `true`  | `string` |           | The name of the regional WAF to associate the API Gateway with |
+| `version`|  `false` | `string` | `Regional`| The AWS Waf version to be used|
```
Knitpick, but `Waf` should be all caps.

MikeSouza commented on lib/index.spec.js:
```diff
@@ -33,6 +38,19 @@ describe('AssociateWafPlugin', () => {
     })
+
+
+    it('should have access to the serverless instance', () => {
```
Remove this test, it's duplicated (copy and pasted?)

MikeSouza commented on lib/index.spec.js:
```diff
@@ -163,45 +180,201 @@ describe('AssociateWafPlugin', () => {
-      await expectOutputRestApiIdToHaveBeenCalled(plugin)
+  })
+
+  describe('with configuration', () => {
```
Just a tip for the future, try to minimize the noise from a diff so that existing unchanged blocks of code aren't mistaken as being "removed" and "added". Reading through the diff of `index.spec.js` was quite noisy and made it way more difficult to easily review what code you added/ modified.

eharris91 replied:
Thanks, thats great feedback. I found most of the reason for the noise due to autoformatting from vscode. Will make sure to turn that off in the future for stuff like this.

MikeSouza commented:
@eharris91 I built and published to NPM a new release of the plugin (`v1.2.0`).  If you could please confirm it's working that would be greatly appreciated. I currently am not setup to properly manually test this at work (we don't use WAFv2).

eharris91 commented:
@MikeSouza I just upgraded to v1.2.0 and the V2 WAF associate/disassociate is working. Have not tried it with the Regional WAF, can you confirm that Regional is still working?

MikeSouza commented:
@eharris91 Great! Thanks for confirming. I've tested that the existing behavior (WAFv1) still works. Good first PR 👍.
